data/tests/tests_framework.txt

```diff
@@ -50,3 +50,115 @@ test "Test-Framework - Conditions Arithmetic and Loops"
 			10 == "test: loopControl"
 			"test: myFirstTestVariable" == 200 - 50
 			400 * 1024 == "test: mySecondTestVariable"
+
+
+test "Test-Framework - UI Keyboard controls"
+	status active
+	description "Tests if the in-game panels can be controlled by keypresses"
+	sequence
+		# Create/inject the savegame and load it.
+		inject "Three Earthly Barges"
+		call "Load First Savegame"
+		# Enter and leave jobs
+		input
+			key j
+		input
+			key d
+		# Enter trading
+		input
+			key t
+		# Enter bank
+		input
+			key b
+		# Enter spaceport
+		input
+			key p
+		# Enter and leave shipyard
+		# Skipped, since Earth (the starting planet) doesn't have a shipyard.
+		#input
+		#	key s
+		#input
+		#	key l
+		# Enter and leave outfitter
+		input
+			key o
+		input
+			key l
+		# Enter hire-crew section
+		input
+			key h
+		# Depart
+		input
+			key d
+
+
+test "Test-Framework - Load and depart"
+	status active
+	description "Test with only a departure. Mostly tests the testframework itself."
+	sequence
+		# Create/inject the savegame and load it.
+		inject "Three Earthly Barges"
+		call "Load First Savegame"
+		# Check startup conditions.
+		assert
+			"ships: Light Freighter" = 3
+		call "Depart"
+		# Re-check startup conditions.
+		assert
+			"ships: Light Freighter" = 3
+
+
+test "Test-Framework - Recursive Calls"
+	status active
+	description "Test-Framework - Test for calls to other functions"
+	sequence
+		apply
+			"test: callLoopControl" = "test: callLoopControl" + 1
+		branch loopBreak
+			"test: callLoopControl" > 10
+		# Use recursive call to get callLoopControl above 10.
+		call "Test-Framework - Recursive Calls"
+		label loopBreak
+
```

Comment:
That depends a bit on the bug that would trigger the issue, but the most likely error-state would be a continuously growing test-call-callstack with "Test-Framework - Recursive Calls" functions continuously recursively calling itself.

I will check if I can add a timeout so that it would fail on the timeout instead of looping forever.

Reply:
I added an initialize part before the test which sets a watchdog/timeout.